statsmodels/base/data.py

```diff
@@ -482,7 +481,7 @@ def _drop_nans(cls, x, nan_mask):
 
     @classmethod
     def _drop_nans_2d(cls, x, nan_mask):
-        if hasattr(x, 'ix'):
+        if isinstance(x, (Series, DataFrame)):
```

Comment:
maybe `hasattr(x 'loc')` to maintain the ducktyping
I don't know whether it matters here or not

Reply:
8 lines up this was changed last week, was missed here.  It should definitely be consistent.

These are very specifically the two classes we want to catch.